File: app/src/main/java/com/aditya/calculator/application/ApplicationActivity.java
package com.aditya.calculator.application;

import android.app.Application;

import com.aditya.calculator.utils.TypefaceUtil;

public class ApplicationActivity extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        TypefaceUtil.overrideFont(getApplicationContext(), "SERIF", "GoogleSans-Regular.ttf"); // font from assets: "assets/fonts/Roboto-Regular.ttf
    }
}

File: app/src/main/java/com/aditya/calculator/application/ApplicationActivity.kt
package com.aditya.calculator.application

import android.app.Application
import android.graphics.Typeface
import java.lang.reflect.Field

class ApplicationActivity : Application() {
    override fun onCreate() {
        super.onCreate()
        try {
            val customFontTypeface = Typeface.createFromAsset(applicationContext.assets, "GoogleSans-Regular.ttf")
            val defaultFontTypefaceField: Field = Typeface::class.java.getDeclaredField("SERIF")
            defaultFontTypefaceField.isAccessible = true
            defaultFontTypefaceField[null] = customFontTypeface
        } catch (e: NoSuchFieldException) {
            e.printStackTrace()
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }
    }
}